airflow/models/taskinstance.py

```diff
@@ -139,6 +139,10 @@
     from airflow.models.operator import Operator
     from airflow.utils.task_group import MappedTaskGroup, TaskGroup
 
+    hybrid_property = property
+else:
+    from sqlalchemy.ext.hybrid import hybrid_property
```

Comment:
Why is the if-else needed?

Reply:
It's a workaround for mypy not understanding the type of hybrid properties - more info here:
https://github.com/python/mypy/issues/4430#issuecomment-781272415